src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.Icu.cs

```diff
@@ -362,14 +362,14 @@ private unsafe bool StartsWithOrdinalIgnoreCaseHelper(ReadOnlySpan<char> source,
 
                 if (source.Length < prefix.Length)
                 {
-                    if (*b >= 0x80)
+                    if (*b >= 0x80 || HighCharTable[*b])
```

Comment:
We cache the char in a local in existing similar places. I assume that it is for good perf. Do we need to do the same here?

Reply:
I don't know but I'll apply the suggestion anyway to follow the same pattern. Thanks!